testHelper/bodyMatchers.go

```diff
@@ -5,41 +5,82 @@ package testHelper
 import (
 	"encoding/json"
 	"fmt"
+	"io"
 	"reflect"
 	"strings"
 	"testing"
 
 	"github.com/apimatic/go-core-runtime/https"
 )
 
-// RawBodyMatcher checks if the expectedBody is contained within the JSON response body.
+func setTestingError(test *testing.T, responseArg any, expectedArg any){
+	test.Errorf("got \n%v \nbut expected %v", responseArg, expectedArg)
+}
+// RawBodyMatcher compares the response body with the expected body via simple string checking. In case of Binary response, byte-by-byte comparison is performed. The expected body takes a URI path to a remote file to compare with the Binary response, which must be valid URI path.
 func RawBodyMatcher[T any](test *testing.T, expectedBody string, responseObject T) {
 	responseBytes, _ := json.Marshal(&responseObject)
 	responseBody := string(responseBytes)
 
 	if !strings.Contains(responseBody, expectedBody) {
-		test.Errorf("got \n%v \nbut expected %v", responseBody, expectedBody)
+		setTestingError(test, responseBody, expectedBody)
 	}
 }
 
-// NativeBodyMatcher compares the JSON response body with the expected JSON body.
-func NativeBodyMatcher[T any](test *testing.T, expectedBody string, responseObject T) {
-	responseBytes, _ := json.Marshal(&responseObject)
-	var expected, response any
-	expectedError := json.Unmarshal([]byte(expectedBody), &expected)
-	responseError := json.Unmarshal(responseBytes, &response)
+// NativeBodyMatcher compares the response body as a primitive type(int, int64, float64, bool & time.Time) using a simple equality test.
+// Response must match exactly except in case of arrays where array ordering and strictness can be controlled via other options.
+func NativeBodyMatcher(test *testing.T, expectedBody string, responseBody io.ReadCloser, isArray, checkArrayCount bool) {
+	
+	responseBytes, responseReadErr := io.ReadAll(responseBody)
+	if responseReadErr != nil {
+		test.Errorf("Invalid response data: %v", responseReadErr)
+	}
+	expectedBytes := []byte(expectedBody)
+	
+	if (isArray){
+		matchNativeArray(test, expectedBytes, responseBytes, checkArrayCount)
+		return
+	}
+	if !reflect.DeepEqual(responseBytes, expectedBytes) {
+		setTestingError(test, string(responseBytes), string(expectedBytes))
+	}
+}
 
-	if expectedError != nil || responseError != nil {
+func matchNativeArray(test *testing.T, expectedBytes, responseBytes []byte, checkArrayCount bool) {
+	var expected, response []any
+	expectedErr := json.Unmarshal(expectedBytes, &expected)
+	responseErr := json.Unmarshal(responseBytes, &response)
+	if expectedErr != nil || responseErr != nil {
 		test.Error("error while unmarshalling for comparison")
 	}
-
+	if (!checkArrayCount){
```

Comment:
Why call `matchNativeArrayCount` when `checkArrayCount` is false? Shouldn't you do it when `checkArrayCount` is true?